compass_app/app/lib/routing/router.dart

```diff
@@ -1,21 +1,35 @@
-import '../ui/results/widgets/results_screen.dart';
 import 'package:go_router/go_router.dart';
 import 'package:provider/provider.dart';
 
 import '../ui/results/view_models/results_viewmodel.dart';
+import '../ui/results/widgets/results_screen.dart';
+import '../ui/search_form/view_models/search_form_viewmodel.dart';
+import '../ui/search_form/widgets/search_form_screen.dart';
 
 /// Top go_router entry point
 final router = GoRouter(
-  initialLocation: '/results',
+  initialLocation: '/',
   routes: [
     GoRoute(
-      path: '/results',
+      path: '/',
       builder: (context, state) {
-        final viewModel = ResultsViewModel(
-          destinationRepository: context.read(),
-        );
-        return ResultsScreen(viewModel: viewModel);
+        final viewModel = SearchFormViewModel(regionRepository: context.read());
+        return SearchFormScreen(viewModel: viewModel);
       },
+      routes: [
+        GoRoute(
+          path: 'results',
+          builder: (context, state) {
+            final viewModel = ResultsViewModel(
+              destinationRepository: context.read(),
+            );
+            final parameters = state.uri.queryParameters;
+            // TODO: Pass the rest of query parameters to the ViewModel
+            viewModel.search(continent: parameters['destination']);
```

Comment:
I have a mix-up between "region" "destination" and "continent" here, I will clean this up in future PRs.